src/librustc_typeck/check/wfcheck.rs

```diff
@@ -118,12 +119,11 @@ impl<'ccx, 'gcx> CheckTypeWellFormedVisitor<'ccx, 'gcx> {
 
                 let trait_ref = ccx.tcx.impl_trait_ref(ccx.tcx.map.local_def_id(item.id)).unwrap();
                 ccx.tcx.populate_implementations_for_trait_if_necessary(trait_ref.def_id);
-                match ccx.tcx.lang_items.to_builtin_kind(trait_ref.def_id) {
-                    Some(ty::BoundSend) | Some(ty::BoundSync) => {}
-                    Some(_) | None => {
-                        if !ccx.tcx.trait_has_default_impl(trait_ref.def_id) {
-                            error_192(ccx, item.span);
-                        }
+                let sync_trait = ccx.tcx.require_lang_item(lang_items::SyncTraitLangItem);
+                let send_trait = ccx.tcx.require_lang_item(lang_items::SendTraitLangItem);
+                if trait_ref.def_id != sync_trait && trait_ref.def_id != send_trait {
```

Comment:
Is this check necessary anymore? AFAIK `trait_has_default_impl` returns true for `Send` and `Sync`.